dbms/src/Server/Server.cpp

```diff
@@ -1410,22 +1410,20 @@ int Server::main(const std::vector<std::string> & /*args*/)
         global_context->setMinMaxIndexCache(minmax_index_cache_size);
 
     /// Size of max memory usage of DeltaIndex, used by DeltaMerge engine.
-    /// This setting is currently a bit tricky:
     /// - In non-disaggregated mode, its default value is 0, means unlimited, and it
-    //    controls the number of total bytes keep in the memory.
-    /// - In disaggregated mode, its default value is 2000. 0 means cache is disabled, and it
-    ///   controls the **number** of trees keep in the memory. <-- Will be fixed.
+    ///   controls the number of total bytes keep in the memory.
+    /// - In disaggregated mode, its default value is memory_capacity_of_host * 0.02.
+    ///   0 means cache is disabled.
     ///   We cannot support unlimited delta index cache in disaggregated mode for now,
     ///   because cache items will be never explicitly removed.
     if (global_context->getSharedContextDisagg()->isDisaggregatedComputeMode())
     {
-        size_t n = config().getUInt64("delta_index_cache_count", 2000);
+        constexpr auto delta_index_cache_ratio = 0.02;
+        size_t n
+            = config().getUInt64("delta_index_cache_size", server_info.memory_info.capacity * delta_index_cache_ratio);
```

Comment:
it will be better to have a backup when memory_capacity_of_host is not obtained.